tasks/common/docker.py

```diff
@@ -8,39 +12,65 @@
 _it = lambda it: '' if it is None else ' -it'  # interactive
 _runtime = lambda runtime: '' if runtime is None else f' --runtime {runtime}'
 _cache_from = lambda cache_from: '' if cache_from is None else f' --cache-from {cache_from}'
+_privileged = lambda privileged: '' if privileged is None else ' --privileged'
+_entrypoint = lambda entrypoint: '' if entrypoint is None else f' --entrypoint {entrypoint}'
+_command = lambda command: '' if command is None else f' {command}'
+_config = lambda config: '' if config is None else f' --config {config}'
+_dockerfile = lambda dockerfile: '' if dockerfile is None else f' -f {dockerfile}'
 listable = lambda func: lambda x: ''.join(map(func, x)) if isinstance(x, (list, tuple)) else func(
     x)  # allow single item or list of multiple items
 _p = listable(lambda p: '' if p is None else (f' -p {p}' if ':' in str(p) else f' -p {p}:{p}'))  # port mapping
 _v = listable(lambda v: '' if v is None else (f' -v {v}' if ':' in str(v) else f' -v {v}:{v}'))  # volume mount
 _e = listable(lambda e: '' if e is None else f' -e {e}')  # environment variable
 
 
-def build(c, dockerfile, tag=None, target=None, host=None, cache_from=None):
-    c.run(
-        f'{_host(host)} docker '
-        f'build{_target(target)}{_tag(tag)}{_cache_from(cache_from)} '
-        f'-f {dockerfile} {ROOT_DIR}'
-    )
+class PipedDockerfile(NamedTuple):
```

Comment:
I think some documentations will be good here

Reply:
around the whole tasks, I meant, and on how developers can use these with inv

Reply:
Done.